### PR TITLE
Replace hardcoded Coveo api keys with  a call to Netlify function

### PR DESCRIPTION
**Pull request by sean-breen**
### Proposed changes

Previously the values of each Coveo API key were hardcoded into the hugo theme code in `layouts/partial/scripts.html` which would pass the appropriate key value to the `js.Build` command. This would result in the api key being embedded in the minified javascript code, and observable via the inspection tools in a web browser.

This change removes those hardcoded values and replaces them with an API call to a Netlify function which will return the value of the `COVEO_SEARCH_TOKEN` variable from the Netlify site's environment variables.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [`CONTRIBUTING`](https://github.com/nginxinc/nginx-hugo-theme/blob/main/CONTRIBUTING.md) document
- [x] If applicable, I have added tests that prove my fix is effective or that my feature works
- [x] If applicable, I have checked that any relevant tests pass after adding my changes
- [x] I have updated any relevant documentation ([`README.md`](https://github.com/nginxinc/nginx-hugo-theme/blob/main/README.md) and [`CHANGELOG.md`](https://github.com/nginxinc/nginx-hugo-theme/blob/main/CHANGELOG.md))

### CODE REVIEW

File: assets/js/coveo.js [hunk 1]
```diff
@@ -1,6 +1,15 @@
...
-    Coveo.SearchEndpoint.configureCloudV2Endpoint("", params.coveokey);
+
+    // call our Netlify function via API to get the coveo search key
+    async function getSearchToken() {
+      const response = await fetch(
+        "netlify/functions/search_token"
+      );
+      return await response.json();
+    }
+
```
Comment (nginx-jack) [inline assets/js/coveo.js:11]: You'll need to `await` the function itself, calling it in line 12 like that I don't think the value will be populated. You could instead have;
`const searchToken = await getSearchToken()`
and update line 12 with 
`Coveo.SearchEndpoint.configureCloudV2Endpoint("", searchToken);`

Comment (sean-breen) [reply assets/js/coveo.js:11]: Could I use the `.then()` clause to handle the promise without using `await` at that level?

**Review by ADubhlaoich — APPROVED** [review]
Approving for the intent of the effort and to ensure any minimum amount of approvals for merges is reached, but I defer to @nginx-jack to decide when the PR is ready to actually merge.

File: assets/js/coveo.js (hunk 1 above)
Comment (nginx-jack) [reply assets/js/coveo.js:11]: We'd a chat offline and made some tweaks. Once it's validated in a dev environment then we should be good to go :)

**Pull request merged**